content/api/templates.apib

```diff
@@ -4,179 +4,131 @@ description: Manage reusable content templates that are run through the SparkPos
+On request failure, an `errors` array will be returned along with HTTP 4xx or 5xx.
+To help debug content issues, templates errors are more detailed than usual.
+Each error is described in a object with the following attributes:
+
++ Data Structure: Attributes
+    + message: (string) - Explains the type of error
+    + code: (string) - Identifies the type of error
+    + description: (string) - Explanation of error
+    + part (enum) - Identifies the content section where the error occurred
+        + text
+        + html
+        + Header:Subject
+        + text/plain
+    + line (number) - Identifies the line number within the MIME part identified by the part attribute
++ Sample
+
+    ```json
+    {
+        "errors": [
             {
-                "name": "Summer Sale!",
-                "shared_with_subaccounts": false,
-                "content": {
-                    "from": "marketing@bounces.company.example",
-                    "subject": "Summer deals",
-                    "html": "<b>Check out these deals!</b>"
-                }
+                "message": "substitution language syntax error in template content",
+                "code": "3000",
+                "description": "Error while compiling part text: line 4: syntax error near 'age'",
+                "part": "html",
+                "line": 4,
             }
-
-+ Response 200 (application/json)
-
-        {
-            "results": {
-                "id": "11806290401558530"
-            }
-        }
-
-+ Response 400 (application/json)
-
-        {
-            "errors": [
-                {
-                    "description": "Unconfigured or unverified sending domain.",
-                    "code": "7001",
-                    "message": "Invalid domain"
-                }
-            ]
-        }
-
-+ Response 400 (application/json)
-
-        {
-            "errors": [
-                {
-                    "description": "Subaccounts cannot set the shared_with_subaccounts flag",
-                    "code": "1200",
-                    "message": "invalid params"
-                }
-            ]
-        }
-
-+ Response 422 (application/json)
-
-        {
-            "errors": [
-                {
-                    "part": "text",
-                    "description": "Error while compiling part text: line 4: syntax error near 'age'",
-                    "line": 4,
-                    "code": "3000",
-                    "message": "substitution language syntax error in template content"
-                }
-            ]
-        }
-
-+ Request Create Parts (application/json)
+        ]
+    }
+    ```
+
+### Create a Template [POST /template]
+
+Templates are created as drafts.
+
++ Data Structure
+    + id (string) - Unique alphanumeric ID used to reference the template. At a minimum, id or name is required upon creation. It is auto-generated if not provided. After a template has been created, this property cannot be changed. Maximum length - 64 bytes
```

Comment:
Make the field names code. id -> `id` and name -> `name`

Reply:
Might not be necessary to say "After a template has been created, this property cannot be changed." since the field won't be listed in the update object.